File: TheMindGamePhp/src/Card.php
<?php

declare(strict_types=1);

namespace App;

final class Card
{
    private int $number;

    public function __construct(int $number)
    {
        $this->number = $number;
    }

    public function number(): int
    {
        return $this->number;
    }

    public function __toString(): string
    {
        return (string)$this->number;
    }
}

File: TheMindGameJava/src/ReadModel/Card.java
package ReadModel;

public class Card implements Comparable<Card> {
    private final int number;

    public Card(int number) {
        this.number = number;
    }

    public int number() {
        return this.number;
    }

    @Override
    public String toString() {
        return "Card(" + number + ')';
    }

    @Override
    public int compareTo(Card o) {
        return this.number >= o.number ? 1 : 0;
    }
}
